### PR TITLE
MGDOBR-1094: Enable the edit button in case of Processor is in Failed state

### PR DESCRIPTION
**Pull request by ppadti**
JIRA: https://issues.redhat.com/browse/MGDOBR-1094

Steps to test it (on chrome):

- Create the Processor with Incorrect parameters to enter into Failed state.
- Check for the Edit button to update the parameters to make it to Ready state.

### CODE REVIEW

**Comment by Srihari1192** [comment]
@ppadti Upon updating the processor from **Failed** state , Processor update status is not changing in UI for either **success** or **Failed** state, Page remains as **Creating**
Screenshot from 2022-10-12 11-57-57

File: src/utils/resourceUtils.tsx [hunk 1]
```diff
@@ -10,3 +10,12 @@ const canDeleteResource = (resourceStatus: ManagedResourceStatus): boolean => {
...
 
 export { canDeleteResource };
+
+const canEditResource = (resourceStatus: ManagedResourceStatus): boolean => {
+  /** It's only possible to edit a resource if it's in the "ready" or "failed" status */
+  return (
+    resourceStatus === ManagedResourceStatus.Ready ||
+    resourceStatus === ManagedResourceStatus.Failed
+  );
+};
```
[the remark this answers is not in the document]
Comment (vpellegrino) [reply src/utils/resourceUtils.tsx:20]: +1 @kelvah's observation

File: src/utils/resourceUtils.tsx [hunk 2]
```diff
@@ -10,3 +10,12 @@ const canDeleteResource = (resourceStatus: ManagedResourceStatus): boolean => {
...
+export { canEditResource };
```
[the remark this answers is not in the document]
Comment (ppadti) [reply src/utils/resourceUtils.tsx]: Yes, Okay.

**Review by vpellegrino — APPROVED** [review]
LGTM, thanks @ppadti

**Review by kelvah — APPROVED** [review]
LGTM :tada:

**Comment by Srihari1192** [comment]
@ppadti @kelvah Please find the observation , not sure this is known issue
When updating Slack processor , `Webhook URL` parameter content is showing as cleared text box , user should be able to see the existing data and able to update `Webhook URL `parameter aswell
Screenshot from 2022-10-12 18-24-26

**Comment by manstis** [comment]
> @ppadti @kelvah Please find the observation , not sure this is known issue When updating Slack processor , `Webhook URL` parameter content is showing as cleared text box , user should be able to see the existing data and able to update `Webhook URL `parameter as well

I believe this is by design. The URL is a "secret" and hence not displayed when editing.

The hint/text below the `TextBox` says as much.

**Comment by Srihari1192** [comment]
@ppadti While Updating Slack Processor details,  

> > @ppadti @kelvah Please find the observation , not sure this is known issue When updating Slack processor , `Webhook URL` parameter content is showing as cleared text box , user should be able to see the existing data and able to update `Webhook URL `parameter as well
> 
> I believe this is by design. The URL is a "secret" and hence not displayed when editing.
> 
> The hint/text below the `TextBox` says as much.

Okay got It , Thanks

**Review by Srihari1192 — APPROVED** [review]

**Comment by kelvah** [comment]
E2E dev tests are failing but the changes in this PR do not affect them (there are no tests there covering the processor editing).

Merging anyway.

**Pull request merged**